go/libraries/doltcore/doltdb/doltdb.go

```diff
@@ -828,6 +828,42 @@ func (ddb *DoltDB) VisitRefsOfType(ctx context.Context, refTypeFilter map[ref.Re
 	})
 }
 
+// GetRefByNameInsensitive searches this Dolt database's branch, tag, and head refs for a case-insensitive
+// match of the specified ref name. If a matching DoltRef is found, it is returned; otherwise an error is returned.
+func (ddb *DoltDB) GetRefByNameInsensitive(ctx context.Context, refName string) (ref.DoltRef, error) {
+	branchRefs, err := ddb.GetBranches(ctx)
```

Comment:
This is fine for now but speaks to a need for constant time case insensitive lookups for these things, rather than this 3N approach